dvc/scm/base.py

```diff
@@ -112,10 +140,12 @@ def brancher(
             revs.extend(tags)
 
         for rev in revs:
-            self.checkout(rev)
+            if checkout:
+                self.checkout(rev)
             yield rev
```

Comment:
Quick question. How should it behave if there are some uncommitted changes to DVC files in the current working space? I think that `dvc fetch -a` and `dvc fetch` will behave differently in this case, and differently from what we have now (do not allow even to run such a command if there are some uncommitted changes). I think it is a design question, and is somewhat related to this https://github.com/iterative/dvc/issues/1633 . I would discuss this with @efiop anyways though, because it might change the current behavior.

Reply:
https://github.com/iterative/dvc/pull/1709#issuecomment-472440054

Reply:
Decided to add the "Working tree" section if SCM has any uncommited changes. Maybe would to limit the "dirty" to `*.dvc` files and stage outs in future.